include/zephyr/net/wifi_nm.h

```diff
@@ -0,0 +1,100 @@
+/** @file
+ * @brief Wi-Fi Network manager API
+ *
+ * This file contains the Wi-Fi network manager API. These APIs are used by the
+ * any network management application to register as a Wi-Fi network manager.
+ */
+
+/*
+ * Copyright (c) Nordic Semiconductor ASA.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_ZEPHYR_NET_WIFI_NM_H_
+#define ZEPHYR_INCLUDE_ZEPHYR_NET_WIFI_NM_H_
+
+#include <zephyr/kernel.h>
+#include <zephyr/types.h>
+#include <zephyr/sys/iterable_sections.h>
+#include <zephyr/net/net_if.h>
+#include <zephyr/net/wifi_mgmt.h>
+/**
+ * @brief Wi-Fi Network manager API
+ * @defgroup wifi_nm Wi-Fi Network Manager API
+ * @ingroup networking
+ * @{
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief WiFi Network manager instance
+ */
+struct wifi_nm_instance {
+	/** Name of the Network manager instance */
+	const char *name;
+	/** Wi-Fi Management operations */
+	const struct wifi_mgmt_ops *ops;
+	/** List of Managed interfaces */
+	struct net_if *mgd_ifaces[CONFIG_WIFI_NM_MAX_MANAGED_INTERFACES];
+};
+
+#define WIFI_NM_NAME(name) wifi_nm_##name
+
+#define DEFINE_WIFI_NM_INSTANCE(_name, _ops)		\
+	static STRUCT_SECTION_ITERABLE(wifi_nm_instance, WIFI_NM_NAME(_name)) = { \
+		.name = STRINGIFY(_name),		\
+		.ops = _ops,				\
+		.mgd_ifaces = { NULL },		\
+	}
+
+/**
+ * @brief Get a Network manager instance for a given name
+ *
+ * @param name Name of the Network manager instance
+ *
+ * @return struct wifi_nm_instance*  Pointer to Network manager instance
```

Comment:
There's no need to document returned type, it won't render nicely. Here and below.